src/openms/source/QC/FragmentMassError.cpp

```diff
@@ -108,6 +108,105 @@ namespace OpenMS
     return theo_spectrum;
   }
 
+  void calculateFME_(PeptideIdentification& pep_id, const MSExperiment& exp, const QCBase::SpectraMap& map_to_spectrum, bool& print_warning, double tolerance, FragmentMassError::ToleranceUnit tolerance_unit, double& accumulator_ppm, UInt32& counter_ppm, WindowMower& window_mower_filter)
+  {
+    if (pep_id.getHits().empty())
+    {
+      OPENMS_LOG_WARN << "PeptideHits of PeptideIdentification with RT: " << pep_id.getRT() << " and MZ: " << pep_id.getMZ() << " is empty.";
+      return;
+    }
+
+    //---------------------------------------------------------------------
+    // FIND DATA FOR THEORETICAL SPECTRUM
+    //---------------------------------------------------------------------
+
+    // sequence
+    const AASequence& seq = pep_id.getHits()[0].getSequence();
+
+    // charge: re-calulated from masses since much more robust this way (PepID annotation of pep_id.getHits()[0].getCharge() could be wrong)
```

Comment:
is this often the case? for which search engines?

Reply:
I just copied existing code. So I really don't know why this is necessary..